core/src/main/scala/org/broadinstitute/dsde/rawls/resourcebuffer/ResourceBufferService.scala

```diff
@@ -0,0 +1,37 @@
+package org.broadinstitute.dsde.rawls.resourcebuffer
+
+import java.util.UUID
+
+import org.broadinstitute.dsde.rawls.dataaccess.resourcebuffer.ResourceBufferDAO
+import org.broadinstitute.dsde.rawls.model.ProjectPoolType.ProjectPoolType
+import org.broadinstitute.dsde.rawls.model.{GoogleProjectId, PoolId, ProjectPoolId, ProjectPoolType, UserInfo}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object ResourceBufferService {
+  def constructor(resourceBufferDAO: ResourceBufferDAO)(userInfo: UserInfo)(implicit executionContext: ExecutionContext): ResourceBufferService = {
+    new ResourceBufferService(resourceBufferDAO, userInfo)
+  }
+}
+class ResourceBufferService(resourceBufferDAO: ResourceBufferDAO, protected val userInfo: UserInfo) {
+
+  def GetGoogleProjectFromRBS(projectPoolType: ProjectPoolType): Future[GoogleProjectId] = getGoogleProjectFromRBS(projectPoolType)
+
+  def getGoogleProjectFromRBS(projectPoolType: ProjectPoolType = ProjectPoolType.Regular): Future[GoogleProjectId] = {
+
+    val projectPoolId: ProjectPoolId = resourceBufferDAO.getProjectPoolId(projectPoolType)
+
+    val handoutRequestId = generateHandoutRequestId(userInfo, projectPoolId)
```

Comment:
we should not be generating random request ids. We should use the workspace id so that if we end up retrying for the same workspace we get the same project and don't leak projects